lib/multipart.js

```diff
@@ -39,15 +39,20 @@
     function initiateUpload () {
       self.post(format('%s?uploads', filename), '', headers, options, function (err, res) {
         if (err) return ws.emit('error', err);
-        try {
-          var body = XML.parse(res.body).InitiateMultipartUploadResult;
+        parseXMLString(res.body, function (err, result) {
+          var body;
+          if (err) {
+            return ws.emit('error', err);
+          }
+          if (!result) {
+            return ws.emit('error', new Error('Could not parse XML string'));
+          }
+          body = result.InitiateMultipartUploadResult;
           if (! body || ! body.UploadId) return ws.emit('error', new Error('Invalid upload id'));
-          ws.uploadId = body.UploadId;
+          ws.uploadId = body.UploadId[0];
```

Comment:
Yes that'd be better, `! body.UploadId` would be false with an empty array. `! body.UploadId.length` would also work.

Thank you.